arduino/libraries/WiFi/src/WiFi.cpp

```diff
@@ -705,6 +720,8 @@ void WiFiClass::handleSystemEvent(system_event_t* event)
         }
       }
 
+      mdns_hostname_set(_hostname);
```

Comment:
What's the difference between `mdns_hostname_set` and `tcpip_adapter_set_hostname`?

Reply:
mdns_hostname_set set the name on the mdns onject the other goes to set the hostname on the lwip tcpip interface

Reply:
maybe the second is useless while mdns is enabled

Reply:
in STA mode mdns not works